metricsaggregator/src/main/java/io/dockstore/metricsaggregator/DoubleStatistics.java

```diff
@@ -0,0 +1,114 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.metricsaggregator;
+
+import java.util.List;
+
+/**
+ * Record that contains statistical information obtained from a list of Doubles.
+ */
+public class DoubleStatistics extends Statistics<Double> {
+    public DoubleStatistics() {
+        super();
+    }
+    
+    /**
+     * Constructor that calculates statistical information from the provided list of data points.
+     * @param dataPoints List of Doubles
+     */
+    public DoubleStatistics(List<Double> dataPoints) {
+        super(dataPoints);
+    }
+
+    public DoubleStatistics(Double minimum, Double maximum, Double average, int numberOfDataPoints) {
+        super(minimum, maximum, average, numberOfDataPoints);
+    }
+
+    /**
+     * Constructor used to create a Statistics object that can be used to calculate weighted averages for non-Statistics objects.
+     * A placeholder value is set for the min and maximum fields
+     * @param average
+     * @param numberOfDataPoints
+     */
+    public DoubleStatistics(double average, int numberOfDataPoints) {
+        super(0d, 0d, average, numberOfDataPoints);
+    }
+
+    /**
+     * Create a new Statistics object from a list of statistics by aggregating the list of statistics
+     * @param statistics
+     * @return
+     */
+    public static DoubleStatistics createFromStatistics(List<DoubleStatistics> statistics) {
+        if (statistics.size() == 1) {
+            return statistics.get(0);
+        }
+
+        DoubleStatistics newStatistics = new DoubleStatistics();
+        newStatistics.setAverage(statistics);
+        newStatistics.setMinimum(statistics);
+        newStatistics.setMaximum(statistics);
+        newStatistics.setNumberOfDataPoints(statistics);
+        return newStatistics;
+    }
+
+    /**
+     * Get the lowest value from the list of data points.
+     * @param dataPoints
+     * @return
+     */
+    @Override
+    public Double calculateMinimum(List<Double> dataPoints) {
+        return dataPoints.stream().mapToDouble(d -> d).min().getAsDouble();
+    }
+
+    /**
+     * Get the highest value from the list of data points.
+     * @param dataPoints
+     * @return
+     */
+    @Override
+    public Double calculateMaximum(List<Double> dataPoints) {
+        return dataPoints.stream().mapToDouble(d -> d).max().getAsDouble();
+    }
+
+    /**
+     * Calculate the average from the list of data points.
+     * @param dataPoints
+     * @return
+     */
+    @Override
+    public Double calculateAverage(List<Double> dataPoints) {
+        return dataPoints.stream().mapToDouble(d -> d).average().getAsDouble();
```

Comment:
Would have thought averaging would be one of the dangerous operations requiring `BigDecimal` or `Money`
https://stackoverflow.com/questions/54278847/averaging-bigdecimals-using-streams-api-collectors